File: android/src/main/java/com/example/flutter_oss_sdk/delegate/FlutterOSSUploadListener.java
package com.example.flutter_oss_sdk.delegate;

import com.example.flutter_oss_sdk.model.FlutterOssUploadResponseFailureModel;
import com.example.flutter_oss_sdk.model.FlutterOssUploadResponseProcessModel;
import com.example.flutter_oss_sdk.model.FlutterOssUploadResponseSuccessModel;

public interface FlutterOSSUploadListener {
    void uploadProgress(FlutterOssUploadResponseProcessModel processModel);

    void uploadSuccess(FlutterOssUploadResponseSuccessModel successModel);

    void uploadFailed(FlutterOssUploadResponseFailureModel failureModel);
}

File: ios/Classes/protocal/FlutterOSSUploadListener.swift
//
//  OSSUploadFileListener.swift
//  flutter_oss_sdk
//
//  Created by kangkang on 2023/2/22.
//

import Foundation

protocol FlutterOSSUploadListener: AnyObject {
    func uploadProgress(processModel: FlutterOssUploadResponseProcessModel) -> Void

    func uploadSuccess(successModel: FlutterOssUploadResponseSuccessModel) -> Void

    func uploadFailed(failureModel: FlutterOssUploadResponseFailureModel) -> Void
}
